Assets/Plugins/Editor/Uplift/Common/VersionRequirements.cs

```diff
@@ -252,79 +250,93 @@ public override string ToString()
     }
 
     // When stub is specified
-    public class LoseVersionRequirement : IVersionRequirement
+    public class LoseVersionRequirement : RangeVersionRequirement
     {
-        public Version stub;
-        private Version limit;
-
         public LoseVersionRequirement(string stub) : this(VersionParser.ParseIncompleteVersion(stub)) { }
-        public LoseVersionRequirement(Version stub)
-        {
-            this.stub = stub;
-            limit = stub.Next;
+        public LoseVersionRequirement(Version stub) : base(stub, stub.Next) { }
+
+        public override IVersionRequirement RestrictTo(IVersionRequirement other) {
+            if(other is BoundedVersionRequirement)
+            {
+                if(IsMetBy((other as BoundedVersionRequirement).lower)) return other;
+            }
+            return base.RestrictTo(other);
         }
 
-        public bool IsMetBy(Version version)
+        public override string ToString()
         {
-            return version >= stub && version < limit;
+            return lower.ToString();
         }
+    }
 
-        public IVersionRequirement RestrictTo(IVersionRequirement other)
+    public class BoundedVersionRequirement : RangeVersionRequirement
+    {
+        public BoundedVersionRequirement(string lowerBound) : this(VersionParser.ParseIncompleteVersion(lowerBound)) { }
+        public BoundedVersionRequirement(Version lowerBound) : base(lowerBound, lowerBound.Next) { }
+
+        public override bool IsMetBy(Version version)
         {
-            if (other is NoRequirement || other is MinimalVersionRequirement)
-            {
-                return other.RestrictTo(this);
-            }
-            else if (other is LoseVersionRequirement)
-            {
-                if (IsMetBy((other as LoseVersionRequirement).stub)) return other;
-                if ((other as LoseVersionRequirement).IsMetBy(stub)) return this;
-            }
-            else if(other is BoundedVersionRequirement)
-            {
-                if (IsMetBy((other as BoundedVersionRequirement).lowerBound)) return other;
-                if ((other as BoundedVersionRequirement).IsMetBy(stub)) return this;
-            }
-            else if(other is ExactVersionRequirement)
+            return version > lower && version < upper;
```

Comment:
I am wondering how this doesn't break the following
`new BoundedVersionRequirement("1.0").IsMetBy("1.0.0.")`

Reply:
1.0.0 > 1.0 in version